### PR TITLE
perf: optimize parser

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

File: lib/llhttp/parser.js [hunk 1]
```diff
@@ -87,16 +80,12 @@ const wasm_on_header_value = (p, at, length) => {
 
 const wasm_on_headers_complete = p => {
   currentParser[kHeaderSize] = 0
-  const versionMajor = inst.exports.llhttp_get_http_major(p)
-  const versionMinor = inst.exports.llhttp_get_http_minor(p)
   const rawHeaders = currentParser[kGetHeaders]()
   const statusCode = inst.exports.llhttp_get_status_code(p)
-  const statusMessage = currentParser[kStatusMessage]
```
Comment (dnlup) [inline lib/llhttp/parser.js:94]: If we are never going to need this, I am in favor. Just asking to be sure.

**Review by dnlup — APPROVED** [review]

File: lib/llhttp/parser.js (hunk 1 above)
Comment (ronag) [reply lib/llhttp/parser.js:94]: I don’t think ao

**Comment by mcollina** [comment]
what are the perf benefits?

**Comment by ronag** [comment]
Pre allocate array
Remove unnecessary calls

**Comment by mcollina** [comment]
Any numbers?

**Comment by ronag** [comment]
current

```js
[1] http - no agent  x 680 ops/sec ±1.46% (80 runs sampled)
[1] http - keepalive x 720 ops/sec ±1.24% (84 runs sampled)
[1] http - keepalive - multiple sockets x 4,699 ops/sec ±2.83% (77 runs sampled)
[1] undici - pipeline x 5,055 ops/sec ±5.12% (79 runs sampled)
[1] undici - request x 5,966 ops/sec ±1.68% (83 runs sampled)
[1] undici - pool - request - multiple sockets x 6,045 ops/sec ±1.35% (84 runs sampled)
[1] undici - stream x 6,269 ops/sec ±1.43% (83 runs sampled)
[1] undici - dispatch x 6,526 ops/sec ±1.10% (83 runs sampled)
[1] undici - noop x 6,556 ops/sec ±0.90% (85 runs sampled)
```

branch
```[1] http - no agent  x 680 ops/sec ±1.60% (79 runs sampled)
[1] http - keepalive x 733 ops/sec ±1.15% (80 runs sampled)
[1] http - keepalive - multiple sockets x 4,009 ops/sec ±3.40% (77 runs sampled)
[1] undici - pipeline x 5,062 ops/sec ±4.38% (80 runs sampled)
[1] undici - request x 5,735 ops/sec ±1.74% (80 runs sampled)
[1] undici - pool - request - multiple sockets x 5,827 ops/sec ±1.67% (79 runs sampled)
[1] undici - stream x 6,176 ops/sec ±1.31% (82 runs sampled)
[1] undici - dispatch x 6,394 ops/sec ±1.03% (83 runs sampled)
[1] undici - noop x 6,521 ops/sec ±1.29% (83 runs sampled)
```

**Pull request closed without merging**